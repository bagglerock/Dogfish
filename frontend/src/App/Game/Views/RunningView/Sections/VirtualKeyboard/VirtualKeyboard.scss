.virtual-keyboard__button {
  &:hover {
    background-color: darkgreen;
    border-color: white;
  }
}
