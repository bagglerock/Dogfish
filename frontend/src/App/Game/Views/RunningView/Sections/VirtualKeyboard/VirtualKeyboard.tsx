import { map, split } from 'lodash';
import React from 'react';
import { Button } from 'react-bootstrap';

const ALPHABET = 'abcdefghijklmnopqrstuvwxyz';

export const VirtualKeyboard: React.FC<KeyboardProps> = ({ updateWord }) => {
  const alphabetArray = split(ALPHABET, '');

  const handleOnClick: React.MouseEventHandler<HTMLButtonElement> = e => {
    const letter = (e.target as HTMLButtonElement).value;

    updateWord(letter);
  };

  return (
    <div className="d-flex flex-wrap game-card m-2">
      {map(alphabetArray, letter => {
        return (
          <Button key={letter} className="p-style-1" onClick={handleOnClick} value={letter.toUpperCase()}>
            {letter.toUpperCase()}
          </Button>
        );
      })}
    </div>
  );
};

interface KeyboardProps {
  updateWord(letter: string): void;
}
