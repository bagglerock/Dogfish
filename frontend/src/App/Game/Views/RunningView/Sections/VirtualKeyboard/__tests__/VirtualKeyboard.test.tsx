import { render } from '@testing-library/react';
import { VirtualKeyboard } from 'app/Game/Views/RunningView/Sections/VirtualKeyboard/VirtualKeyboard';

test('renders VirtualKeyboard', () => {
  const { container } = render(<VirtualKeyboard updateWord={jest.fn()} />);

  expect(container).toMatchSnapshot();
});
