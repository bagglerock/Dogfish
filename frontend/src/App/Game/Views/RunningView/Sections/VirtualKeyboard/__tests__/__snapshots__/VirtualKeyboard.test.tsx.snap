// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders VirtualKeyboard 1`] = `
<div>
  <div
    class="d-flex flex-wrap game-card m-2"
  >
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="A"
    >
      A
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="B"
    >
      B
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="C"
    >
      C
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="D"
    >
      D
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="E"
    >
      E
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="F"
    >
      F
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="G"
    >
      G
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="H"
    >
      H
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="I"
    >
      I
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="J"
    >
      J
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="K"
    >
      K
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="L"
    >
      L
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="M"
    >
      M
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="N"
    >
      N
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="O"
    >
      O
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="P"
    >
      P
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="Q"
    >
      Q
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="R"
    >
      R
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="S"
    >
      S
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="T"
    >
      T
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="U"
    >
      U
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="V"
    >
      V
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="W"
    >
      W
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="X"
    >
      X
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="Y"
    >
      Y
    </button>
    <button
      class="p-style-1 btn btn-primary"
      type="button"
      value="Z"
    >
      Z
    </button>
  </div>
</div>
`;
